src/oci/conmon.rs

```diff
@@ -0,0 +1,292 @@
+//! Interface to [conmon][0], the OCI container runtime monitor.
+//!
+//! [0]: https://github.com/containers/conmon
+
+#![allow(dead_code)]
+
+use anyhow::{Context, Result};
+use async_trait::async_trait;
+use derive_builder::Builder;
+use dyn_clone::{clone_trait_object, DynClone};
+use getset::{Getters, Setters};
+use log::LevelFilter;
+use std::{
+    fmt::{self, Debug},
+    path::{Path, PathBuf},
+    process::Output,
+    string::ToString,
+    time::Duration,
+};
+use strum::AsRefStr;
+use tokio::process::Command;
+
+#[derive(Builder, Debug, Getters, Setters)]
+#[builder(pattern = "owned", setter(into))]
+// Conmon is the main structure to be used when interacting with the container monitor.
+pub struct Conmon {
+    #[getset(get, set)]
+    #[builder(private, default = "Box::new(DefaultExecCommand)")]
+    /// The executor for conmon
+    exec: Box<dyn ExecCommand>,
+
+    #[get]
+    /// Path to the conmon binary
+    binary: PathBuf,
+}
+
+impl Conmon {
+    /// Run conmon with the provided args and return the output if the command execution succeeds.
+    /// This can still mean that conmon itself failed, which can be verified via the exist status
+    /// of the output.
+    pub async fn run(&self, args: &[Arg]) -> Result<Output> {
+        self.exec().run_output(self.binary(), args).await
+    }
+}
+
+#[derive(Clone, Default, Debug)]
+/// DefaultExecCommand is a wrapper which can be used to execute conmon in a standard way.
+struct DefaultExecCommand;
+
+impl ExecCommand for DefaultExecCommand {}
+
+#[async_trait]
+trait ExecCommand: Debug + DynClone + Send + Sync {
```

Comment:
interesting note here, if we are going to exec out to anything else this could really be used by the other things. Specifically I'm thinking `runc` here, not just hypothetically. Would we want to move this into it's own module?

Reply:
Yes, we have it in multiple places and I having a general ExecCommand trait seems useful to me. :+1:

Reply:
Now that I brought this up, I was just thinking that with the `runc` i'm going to have to add support for "sub" commands. aka `runc create [args]` vs `runc delete [args]` so now that I have said that, it's probably best if I generalize it in my PR updated to support sub commands...
I am initially thinking of something like....
`async fn run_output(&self, binary: &Path, cmds: &[SubCmd], args: &[Arg]) -> Result<Output>` supports multiple sub commands... though probably not needed for runc. anyhow. just a thought.

Reply:
I think sub-commands are just args in general, right?